aten/src/ATen/native/Distributions.cpp

```diff
@@ -197,6 +197,118 @@ Tensor& geometric_(Tensor& self, double p, Generator* gen) {
+  const Vec n1 = radius * Vec(costheta);
+  const Vec n2 = radius * Vec(sintheta);
+  vec256::fmadd(n1, std, mean).store(data);
+  vec256::fmadd(n2, std, mean).store(data + 8);
+}
+
+void normal_fill_AVX2(Tensor& self, const float mean, const float std, Generator* gen) {
+  float *data = self.data_ptr<float>();
+  auto size = self.numel();
+  CPUGenerator* generator = get_generator_or_default<CPUGenerator>(gen, detail::getDefaultCPUGenerator());
+  std::lock_guard<std::mutex> lock(generator->mutex_);
+  for (int64_t i = 0; i < size; ++i) {
+    at::uniform_real_distribution<float> uniform(0, 1);
+    data[i] = uniform(generator);
+  }
+
+  using Vec = Vec256<float>;
+  const Vec two_pi = Vec(2.0f * M_PI);
+  const Vec one = Vec(1.0f);
+  const Vec minus_two = Vec(-2.0f);
+  const Vec mean_v = Vec(mean);
+  const Vec std_v = Vec(std);
+
+  for (int64_t i = 0; i < size - 15; i += 16) {
+    normal_fill_16_AVX2(data + i, two_pi, one, minus_two, mean_v, std_v);
+  }
+
+  if (size % 16 != 0) {
+    // Recompute the last 16 values.
+    data = data + size - 16;
+    for (int64_t i = 0; i < 16; ++i) {
+      at::uniform_real_distribution<scalar_t> uniform(0, 1);
+      data[i] = uniform(generator);
+    }
+    normal_fill_16_AVX2(data, two_pi, one, minus_two, mean_v, std_v);
+  }
+}
+#else
+void normal_fill_AVX2(Tensor& self, const float mean, const float std, Generator* gen) {
+  TORCH_CHECK(false, "normal_fill_AVX2: not compiled with AVX2 support");
+}
+#endif
+
+Tensor& normal_cpu_(Tensor& self, double mean, double std, Generator* gen) {
+  TORCH_CHECK(std > 0.0, "normal_ expects std > 0.0, but found std=", std);
+  bool normal_avx2 = false;
+#ifdef __AVX2__
+  if (self.scalar_type() == ScalarType::Float && size = self.numel() > 16 && self.is_contiguous())
+      normal_avx2 = true;
+#endif
+  if (normal_avx2) {
+    normal_fill_AVX2(self, static_cast<float>(mean), static_cast<float>(std), gen);
+  } else {
+    normal_stub(kCPU, self, mean, std, gen);
+  }
+  return self;
+}
+
+Tensor& normal_out_cpu(Tensor& output, const Tensor& mean, double std, Generator* gen) {
+  normal_cpu_(output, 0, std, gen);
+  output.add_(mean);
+  return output;
+}
+
+Tensor& normal_out_cpu(Tensor& output, double mean, const Tensor& std, Generator* gen) {
+  normal_cpu_(output, 0, 1, gen);
+  auto mean_tensor = at::full({}, mean, output.options());
+  output.mul_(std).add_(mean_tensor);
+  return output;
+}
+
+Tensor& normal_out_cpu(Tensor& output, const Tensor& mean, const Tensor& std, Generator* gen) {
+  normal_cpu_(output, 0, 1, gen);
+  output.mul_(std).add_(mean);
+  return output;
+}
+
+Tensor normal_cpu(const Tensor& mean, double std, Generator* gen) {
+  Tensor ret = at::empty_like(mean, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
+  normal_out_cpu(ret, mean, std, gen);
+  return ret;
+}
+
+Tensor normal_cpu(double mean, const Tensor& std, Generator* gen) {
+  Tensor ret = at::empty_like(std, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
```

Comment:
@VitalyFedyunin When we do ports like this, should we really be continuing to use the `LEGACY_CONTIGUOUS_MEMORY_FORMAT` here?

Reply:
Agree, this is unnecessary and we should omit this argument or use MemoryFormat::Contiguous when it is required to have contiguous tensor.

In this particular case `ret` is expected to be Contiguous so `MemoryFormat::Contiguous` is required.